### ISSUE TITLE
replace postgrey with postfwd

### ISSUE THREAD

**Issue by gfa** [mail-in-a-box/mailinabox#765]
Hello

I see you patch postgrey to avoid greylisting senders on dnswl.org, postfwd does that by default and more, gmail, hotmail, etc are not greylisted, postfwd will greylist the sender if it is in at least once black list, if it uses a dymanic ip, if helo does not match domain, etc.

Also it happened to me more than once that postgrey hangs on medium traffic servers

i would push a PR if there is interest using postfwd

**Comment by JoshData** [comment]
Yes, moving to a system that we don't have to patch would be great. If it's easy to install & configure. How easy is it?

See #574 for the last time someone suggested it.

**Comment by gfa** [comment]
It is not much more difficult than postgrey, once it is installed you have to call it from postfix the same way as postgrey, the bad thing is that you still use postgrey to do greylisting when appropriate.

sorry I didn't see the old issue, i will push a PR but not before next week.

**Comment by JoshData** [comment]
> the bad thing is that you still use postgrey to do greylisting when appropriate

I'm confused. We still need postgrey with postfwd?

**Comment by gfa** [comment]
On Sat, Mar 26, 2016 at 05:54:26AM -0700, Joshua Tauberer wrote:
> > the bad thing is that you still use postgrey to do greylisting when appropriate
> 
> I'm confused. We still need postgrey with postfwd?

yes, postfwd is a policy front end, it will process emails accordind to
a policy and either whitelist/pass, blacklist reject or greylist it.

greylist is implemented passing the email to postgrey (or another
greylist daemon)

i have an early patch adding postfwd but documentation is not finished
neither test all the use cases

-- 
1AE0 322E B8F7 4717 BDEA BF1D 44BB 1BA7 9F6C 6333

keybase: http://keybase.io/gfa

**Comment by JoshData** [comment]
I see, so we can drop my patch but we keep postgrey. (That's still good.)

**Comment by TabTwo** [comment]
This is basically what I use.
Checks some Whiste/Blacklists ... 

http://ftp.uni-kl.de/pub/linux/gentoo-portage/mail-filter/postfwd/files/postfwd2.example.cf

**Comment by JoshData** [comment]
Cool that's helpful, thanks.
